artistools/lightcurve.py

```diff
@@ -164,6 +171,18 @@ def get_magnitudes(modelpath):
     return filters_dict
 
 
+def bolometric_magnitude(modelpath, timearray):
+    magnitudes = []
+    for timestep, time in enumerate(timearray):
+        spectrum = spectra.get_spectrum(modelpath, timestep, timestep)
+
+        integrated_flux = np.trapz(spectrum['f_lambda'], spectrum['lambda_angstroms'])
+        integrated_luminosity = abs(integrated_flux * -4 * np.pi * np.power(3.086e24, 2))
+        magnitude = 4.74 - (2.5 * np.log10(integrated_luminosity / 3.828e33))
+        magnitudes.append(magnitude)
+    return magnitudes
```

Comment:
To make the units more clear, I suggest replacing 3.087e24 with u.Mpc.to('cm') and 3.828e33 with const.L_sun.to('erg/s').value. 

Using np.trapz here a good choice. I'll remove my function at.spectra.integrate_flux(), which does the same thing in a less-accurate way.

I guess you used the abs function here because your got the right number, only with a negative sign because spectrum['lambda_angstroms'] is in decreasing. So I think it would be better to just put a minus sign here instead of abs(), so that if you really do get a negative answer for the luminosity you will know something has gone wrong!